sponsorship_compassion/models/res_partner.py

```diff
@@ -368,12 +368,20 @@ def forget_me(self):
             if "failure" in message.state:
                 answer = message.get_answer_dict()
                 if isinstance(answer, dict):
-                    error_message = answer.get(
-                        "DataProtection Error", message.failure_reason
+                    error_info = answer.get("Error", {})
+                    error_message = error_info.get("ErrorMessage", "Erreur inconnue")
+
+                    # Format user-friendly error message
+                    user_error_message = (
+                        f"an error occurred during data anonymisation.\n\n"
+                        f"Message : {error_message}"
                     )
                 else:
-                    error_message = message.failure_reason
-                raise UserError(error_message)
+                    user_error_message = (
+                        f"An internal error prevented the data from being anonymised.\n"
+                        f"details: {message.failure_reason}"
+                    )
```

Comment:
An error needs to be logged on the server, you can log it by using the logger provided by Odoo
`_logger.error(...)`